src/fvmcts/fv_mcts_vanilla.jl

```diff
@@ -349,15 +349,15 @@ end
     mdp = planner.mdp
     P = typeof(mdp)
     @assert P <: JointMDP
-    #SV = statetype(P)
+    SV = statetype(P)
     #@assert typeof(SV) <: AbstractVector
     AV = actiontype(P)
     @assert typeof(A) <: AbstractVector
     @req discount(::P)
     @req isterminal(::P, ::SV)
     @subreq insert_node!(planner.tree, planner, s)
     @subreq estimate_value(planner.solved_estimate, mdp, s, depth)
-    @req gen(::DDNOut{(:sp, :r)}, ::P, ::SV, ::A, ::typeof(planner.rng))
+    @req gen(::P, ::SV, ::A, ::typeof(planner.rng)) # XXX this is not exactly right - it could be satisfied with transition
```

Comment:
Not sure I see the issue here. Don't we require a generative model to be implemented?

Reply:
MCTS.jl has the same thing 😛. @zsunberg might know the exact reason for the comment.